xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-default/src/main/java/org/xwiki/security/authentication/internal/DefaultResetPasswordManager.java

```diff
@@ -145,72 +157,68 @@ private boolean checkUserReference(UserReference userReference) throws ResetPass
+        if (userInformation.canUserResetPassword()) {
+            DocumentUserReference documentUserReference = (DocumentUserReference) userReference;
+            DocumentReference reference = documentUserReference.getReference();
+            XWikiContext context = this.contextProvider.get();
 
-                    String saveComment =
-                        this.localizationManager.getTranslationPlain("xe.admin.passwordReset.versionComment");
-                    context.getWiki().saveDocument(userDocument, saveComment, true, context);
+            try {
+                XWikiDocument userDocument = context.getWiki().getDocument(reference, context);
 
-                    return new DefaultResetPasswordRequestResponse(userReference, verificationCode);
-                } catch (XWikiException e) {
-                    throw new ResetPasswordException(
-                        "Error when reading user document to perform reset password request.",
-                        e);
+                if (userDocument.getXObject(LDAP_CLASS_REFERENCE) != null) {
+                    String exceptionMessage =
+                        this.localizationManager.getTranslationPlain("xe.admin.passwordReset.error.ldapUser",
+                            userReference.toString());
+                    throw new ResetPasswordException(exceptionMessage);
                 }
-            } else {
-                // In case the mail is not configured, we log a message to the admin.
-                this.logger.info("User [{}] asked to reset their password, but did not have any email configured.",
-                    userReference);
+
+                BaseObject xObject = userDocument.getXObject(ResetPasswordRequestClassDocumentInitializer.REFERENCE,
+                    true, context);
+                String verificationCode = context.getWiki().generateRandomString(30);
+                xObject.set(ResetPasswordRequestClassDocumentInitializer.VERIFICATION_FIELD,
+                    verificationCode, context);
+                xObject.setDateValue(ResetPasswordRequestClassDocumentInitializer.REQUEST_DATE_FIELD, new Date());
+                String saveComment =
+                    this.localizationManager.getTranslationPlain("xe.admin.passwordReset.versionComment");
+                context.getWiki().saveDocument(userDocument, saveComment, true, context);
+
+                return new DefaultResetPasswordRequestResponse(userReference, verificationCode);
+            } catch (XWikiException e) {
+                throw new ResetPasswordException(
+                    "Error when reading user document to perform reset password request.",
+                    e);
             }
+        } else if (userInformation.userExists && userInformation.userEmail == null) {
+            // In case the mail is not configured, we log a message to the admin.
+            this.logger.info("User [{}] asked to reset their password, but did not have any email configured.",
```

Comment:
Shouldn't this information be communicated to the user requesting for a password reset?

Reply:
Note that I haven't changed the behaviour here, it's just that the diff is not easy to read. So I don't want to change that now. AFAIR we decided to always display the minimum of information when trying to reset a password so it basically always say the same thing to prevent any kind of data leak. 
We could decide to change that but IMO it deserves a dedicated ticket.